pkg/snapshot/publisher.go

```diff
@@ -69,86 +78,107 @@ func (p *Publisher) PublishHeader(header *types.Header) (int64, error) {
  								RETURNING id`,
 		header.Number.Uint64(), header.Hash().Hex(), header.ParentHash.Hex(), headerNode.Cid().String(), "0", p.db.NodeID, "0", header.Root.Hex(), header.TxHash.Hex(),
 		header.ReceiptHash.Hex(), header.UncleHash.Hex(), header.Bloom.Bytes(), header.Time, mhKey, 0).Scan(&headerID)
+
 	return headerID, err
 }
 
 // PublishStateNode writes the state node to the ipfs backing datastore and adds secondary indexes in the state_cids table
-func (p *Publisher) PublishStateNode(node Node, headerID int64) (int64, error) {
+func (p *Publisher) PublishStateNode(node *node, headerID int64, tx *sqlx.Tx) (int64, error) {
 	var stateID int64
 	var stateKey string
-	if !bytes.Equal(node.Key.Bytes(), nullHash.Bytes()) {
-		stateKey = node.Key.Hex()
-	}
-	tx, err := p.db.Beginx()
-	if err != nil {
-		return 0, err
+	if !bytes.Equal(node.key.Bytes(), nullHash.Bytes()) {
+		stateKey = node.key.Hex()
 	}
+
 	defer func() {
-		if p := recover(); p != nil {
+		if rec := recover(); rec != nil {
 			shared.Rollback(tx)
-			panic(p)
-		} else if err != nil {
-			shared.Rollback(tx)
-		} else {
-			err = tx.Commit()
+			panic(rec)
 		}
 	}()
-	stateCIDStr, err := shared.PublishRaw(tx, ipld.MEthStateTrie, multihash.KECCAK_256, node.Value)
+
+	stateCIDStr, mhKey, err := shared.PublishRaw(tx, ipld.MEthStateTrie, multihash.KECCAK_256, node.value)
 	if err != nil {
 		return 0, err
 	}
-	mhKey, _ := shared.MultihashKeyFromCIDString(stateCIDStr)
+
 	err = tx.QueryRowx(`INSERT INTO eth.state_cids (header_id, state_leaf_key, cid, state_path, node_type, diff, mh_key) VALUES ($1, $2, $3, $4, $5, $6, $7)
- 									ON CONFLICT (header_id, state_path) DO UPDATE SET state_path = state_cids.state_path
+ 									ON CONFLICT (header_id, state_path) DO UPDATE SET (state_leaf_key, cid, node_type, diff, mh_key) = ($2, $3, $5, $6, $7)
  									RETURNING id`,
-		headerID, stateKey, stateCIDStr, node.Path, node.NodeType, false, mhKey).Scan(&stateID)
+		headerID, stateKey, stateCIDStr, node.path, node.nodeType, false, mhKey).Scan(&stateID)
+
+	p.currBatchSize += 2
```

Comment:
Should this be `++` instead of `+= 2`? `PublishRaw` also increments it, so we don't need to account for that stmt here.

Reply:
`shared.PublishRaw` we are using here is from `geth` library, and `publishRaw` where we were incrementing `currBatchSize` was locally implemented, updated the code to use only `shared.PublishRaw`.